### PR TITLE
PorousFlow relative permeability

### LINKED ISSUE #7877: Update PorousFlow relative permeabilities
### Description of the enhancement or error report
The Corey relative permeability material available in the `porous_flow` module is incomplete (it doesn't have the residual values), and the popular van Genuchten formulation has not yet been included.

### Rationale for the enhancement or information for reproducing the error
Adds important features and a new formulation to the relative permeability materials. 

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
Added capability

### PR DESCRIPTION
**Pull request by cpgr**
- Refactor PorousFlowRelativePermeabilityBase
- Adds residual saturations to Corey relative permeability
- Adds van Genuchten relative permeability formulation
- Adds tests of all relative permeability relations
- Includes a python script to generate plots of relative permeability vs saturation for each of the tests for including in docs

Closes #7877

### CODE REVIEW

**Comment by cpgr** [comment]
The failed modules test is an unrelated timeout - the tests that are included in this PR pass

File: modules/porous_flow/doc/tests/relperm.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    seff = np.clip((s - sr) / (sls - sr), 0, 1)
+
+    # The liquid relative permeability is
+    relperm = np.sqrt(seff) * (1 - np.power(1 - np.power(seff, 1.0/m), m))**2
+
+    # Relative permeability is then
+    return relperm
+
+# Saturation of phase 0 varies linearly from 0 to 1
+s0 = np.linspace(0,1,200)
```
Comment (andrsd) [inline modules/porous_flow/doc/tests/relperm.py]: spaces after commas

File: modules/porous_flow/doc/tests/relperm.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    return relperm
+
+# Analytical form of van Genuchten's relative permeability
+def vg(s, sr, sls, m) :
+
+    # Effective saturation
+    seff = np.clip((s - sr) / (sls - sr), 0, 1)
+
+    # The liquid relative permeability is
+    relperm = np.sqrt(seff) * (1 - np.power(1 - np.power(seff, 1.0/m), m))**2
```
Comment (andrsd) [inline modules/porous_flow/doc/tests/relperm.py]: spaces around `/`

File: modules/porous_flow/doc/tests/relperm.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    # Effective saturation
+    seff = np.clip((s - sr) / (1.0 - sum_s_r), 0, 1)
+
+    # Relative permeability is then
+    relperm = np.power(seff, n);
+
+    return relperm
+
+# Analytical form of van Genuchten's relative permeability
+def vg(s, sr, sls, m) :
```
Comment (andrsd) [inline modules/porous_flow/doc/tests/relperm.py]: no space between `)` and `:`

**Review by andrsd — CHANGES_REQUESTED** [review]

File: modules/porous_flow/doc/tests/relperm.py (hunk 1 above)
Comment (cpgr) [inline modules/porous_flow/doc/tests/relperm.py]: All fixed. Thanks again @andrsd

**Review by andrsd — APPROVED** [review]

**Pull request merged**